### PR TITLE
Screen reader should read alert when sending empty message or offline

### LINKED ISSUE #4623: Error message is not displayed after clicking ‘Send’ button on without entering value in 'Send a message' edit field.
### Is it an issue related to Adaptive Cards?

No.

### What is the PWD impact?

Users who depend on keyboard will face difficulty as Error message is not displayed after clicking ‘Send’ button on without entering value in 'Send a message' edit field.

### What browsers and screen readers do this issue affect?

Windows: Edge with Windows Narrator

### Are there any code-based customization done to Web Chat?

No, I am using Web Chat without any customizations except "styleOptions".

### What version of Web Chat are you using?

Latest production

### Which area does this issue affect?

Send box

### What is the public URL for the website?

https://web.powerva.microsoft.com/environments/31c9fd16-d6ad-4925-8549-2d578a71b990/bots/a619e4ca-7ccb-4da9-a2c6-48fcba807887/canvas

### How to reproduce the issue?

1. Open the URL https://web.powerva.microsoft.com/environments/31c9fd16-d6ad-4925-8549-2d578a71b990/bots/a619e4ca-7ccb-4da9-a2c6-48fcba807887/canvas in Edge Browser.
2. Microsoft Power Platform home page gets displayed.
3. Navigate to Type your message edit field in ‘Device bot virtual agent’ chat dialog.
4. Do not provide any data in Type your message edit field and navigate to Send button and invoke it.
5. Verify whether error message is displayed or not after invoking ‘Send’ button.

### What do you expect?

Error message should be displayed after clicking ‘Send’ button on without entering value in 'Send a message' edit field.
**Ex:** After clicking 'Send' button on without entering value in type your message edit field, Error message should display as "Please type as message to continue"
**Ensure that** one the error message is displayed screen reader is narrating the message displayed.

### What actually happened?

Error message is not displayed after clicking ‘Send’ button on without entering value in 'Send a message' edit field.
**Current Observation:** After clicking send button on without entering value in type your message edit field, Error message is not displayed.

### Do you have any screenshots or recordings to repro the issue?

_No response_

### Did you find any DOM elements that might have caused the issue?

_No response_

### MAS reference

https://aka.ms/MAS3.3.1

### WCAG reference

_No response_

### WAI-ARIA reference

_No response_

### Adaptive Card JSON

_No response_

### Additional context

Copied from https://msazure.visualstudio.com/CCI/_workitems/edit/16393475.

### PR DESCRIPTION
**Pull request by compulim**
> Fixes #4623.

### Fixed

-  Fixes [#4623](https://github.com/microsoft/BotFramework-WebChat/issues/4623). Screen reader should read error when failed to send an empty message or offline, by [@compulim](https://github.com/compulim) in PR [#4637](https://github.com/microsoft/BotFramework-WebChat/pull/4637)

## Description

When the user press <kbd>ENTER</kbd> or click on the "Send" button, while leaving the message empty or offline, Web Chat will not send the message.

However, we should also alert the user about their mistakes via screen reader.

This PR adds an alert when sending empty message or offline.

## Design

We are adding an assertive live region inside the send box. It will hold the error message.

The error message will be triggered by three UIs: single line text box, multiline text area, and send button.

Since 3 different UIs need to show the error message, we are adding a new `<SendBoxComposer>` to handle the coordination. The new `<SendBoxComposer>` should also handle interactions related to the send box.

However, due to time constraints, we cannot fully complete the `<SendBoxComposer>`. We will complete it when we are working on the customizable send box work (#1839).

## Specific Changes

- Added a new `<SendBoxComposer>`
   - It includes an assertive live region, which can be repetitively read error messages
- Moved submit logic from send button and text box submit, into the new `useSubmit()` hook exposed by the new `<SendBoxComposer>`

-  [x] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  [x] I have updated documentation

## Review Checklist

> This section is for contributors to review your work.

-  [x] Accessibility reviewed (tab order, content readability, alt text, color contrast)
-  [x] ~Browser and platform compatibilities reviewed~
-  [x] CSS styles reviewed (minimal rules, no `z-index`)
-  [x] ~Documents reviewed (docs, samples, live demo)~
-  [x] ~Internationalization reviewed (strings, unit formatting)~
-  [x] ~`package.json` and `package-lock.json` reviewed~
-  [x] ~Security reviewed (no data URIs, check for nonce leak)~
-  [x] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: packages/component/src/providers/internal/SendBox/SendBoxComposer.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    [apiSubmitSendBox, focusRef, scrollToEndRef, setErrorRef, submitErrorRef, timeoutRef]
+  );
+
+  useEffect(
+    // Prevent `setTimeout()` from firing after unmount.
+    () => () => {
+      setErrorRef.current = undefined;
+    },
+    [setErrorRef]
+  );
```
Comment (beyackle2) [inline packages/component/src/providers/internal/SendBox/SendBoxComposer.tsx:124]:
```suggestion
  useEffect(
    // Prevent `setTimeout()` from firing after unmount.
    () => {
      setErrorRef.current = undefined;
    },
    [setErrorRef]
  );
```

File: packages/component/src/providers/internal/SendBox/SendBoxComposer.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    <SendBoxContext.Provider value={context}>
+      {children}
+      <span
+        className={classNames('webchat__submit-error-message', submitErrorMessageClassName)}
+        // "id" is required for "aria-errormessage" as IDREF.
+        // eslint-disable-next-line react/forbid-dom-props
+        id={submitErrorMessageId}
+        role="alert"
+      >
+        {(error && errorMessageStringMap.get(error)) || ''}
```
Comment (beyackle2) [inline packages/component/src/providers/internal/SendBox/SendBoxComposer.tsx:144]:
```suggestion
        {error ? errorMessageStringMap.get(error) : ''}
```

**Review by beyackle2 — CHANGES_REQUESTED** [review]
Mostly looks good, but I have some small suggestions.

File: packages/component/src/providers/internal/SendBox/SendBoxComposer.tsx (hunk 2 above)
Comment (compulim) [reply packages/component/src/providers/internal/SendBox/SendBoxComposer.tsx:144]: I thought there are slim chances that the `errorMessageStringMap` does not have the value.

But as I am using the result as React children node, this is actually fine for `undefined`.

I am taking your suggestions. Thanks.

File: packages/component/src/providers/internal/SendBox/SendBoxComposer.tsx (hunk 1 above)
Comment (compulim) [reply packages/component/src/providers/internal/SendBox/SendBoxComposer.tsx:124]: This is actually intended.

We want to set `setErrorRef` after *unmount*, not *mount*. So the double `() => () =>` will be called during unmount.

**Review by beyackle2 — APPROVED** [review]

**Pull request merged**